pycbc/frame/frame.py

```diff
@@ -600,7 +600,7 @@ def update_cache_by_increment(self, blocksize):
                 n = int(pattern[int(pattern.index('GPS') + 3)])
                 pattern = pattern.replace('GPS%s' % n, str(s)[0:n])
 
-            name = '%s/%s-%s-%s.gwf' % (pattern, self.beg, s, self.dur)
+            name = f'{pattern}/{self.beg}-{s}-{self.dur}.gwf'
```

Comment:
Huh, didn't know you could do this. Interesting!

Reply:
They are called f-strings, I think they are a relatively recent feature.

Reply:
Added in python 3.7 and *very* useful

Reply:
So, perfectly fine now that we require >= 3.7, neat

Reply:
I'm always wondering how far it is worth trying to learn the 4th (or is it 5th) python string formatting scheme given that I don't seem to have needed anything beyond % yet